src/app/storage.js

```diff
@@ -28,12 +33,16 @@ function Storage () {
 
   this.remove = function (name) {
     window.localStorage.removeItem('sol:' + name)
+    return true
   }
 
   this.rename = function (originalName, newName) {
     var content = this.get(originalName)
-    this.set(newName, content)
+    if (!this.set(newName, content)) {
```

Comment:
As discussed, `set()` operates in the same manner - overwrites if it exists. In order to make it consistent, that would have to change too.

I think I would postpone this one item until the next step when we see which layer it makes the most sense in. Happy to change it if it turns out to be useful here.